### ISSUE TITLE
Rest endpoint not working

### ISSUE THREAD

**Issue by yadavsudhir405** [beryx/badass-jlink-plugin#145]
Hi,

I was following up spring-boot example to build image using  this plugin. Image is getting built properly but somehow the endpoint which is exposed in my code is not working. It's giving 404. However when I run the application using a conventional way it works fine.

While running the image I don't get any error message or any such information that can give me some clue about the issues.

This is how my build.gradle file looks:
```plugins {
    id 'java'
    id 'org.springframework.boot' version '2.3.1.RELEASE'
    id 'io.spring.dependency-management' version '1.0.9.RELEASE'
    id "org.beryx.jlink" version "2.21.0"
    id "org.javamodularity.moduleplugin" version "1.6.0"
}
repositories {
    gradlePluginPortal()
    jcenter()
}

sourceCompatibility = JavaVersion.VERSION_14
targetCompatibility = JavaVersion.VERSION_14
tasks.withType(JavaCompile) {
    options.encoding = 'UTF-8'
}

configurations {
    compileOnly {
        extendsFrom annotationProcessor
    }
}

jar {
    enabled = true;
}
application{
    mainModule='com.sudhir.registration'
}

configurations {
    springFactoriesHolder { transitive = false }
}

dependencies {
    springFactoriesHolder 'org.springframework.boot:spring-boot-actuator-autoconfigure'
    springFactoriesHolder 'org.springframework.boot:spring-boot-autoconfigure'
    springFactoriesHolder 'org.springframework.boot:spring-boot'

    implementation 'org.springframework.boot:spring-boot'
    implementation 'org.springframework.boot:spring-boot-starter-web'
    implementation 'org.springframework.boot:spring-boot-starter-actuator'

    testImplementation('org.springframework.boot:spring-boot-starter-test') {
        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
    }
}

mainClassName = "com.sudhir.registration.ApiApplication"


prepareMergedJarsDir.doLast {
    // extract and merge META-INF/spring.factories from springFactoriesHolder
    def factories = configurations.springFactoriesHolder.files.collect {
        def props = new Properties()
        props.load(zipTree(it).matching { include 'META-INF/spring.factories' }.singleFile.newInputStream())
        props
    }
    def mergedProps = new Properties()
    factories.each { props ->
        props.each { key, value ->
            def oldVal = mergedProps[key]
            mergedProps[key] = oldVal ? "$oldVal,$value" : value
        }
    }
    def content = mergedProps.collect { key, value ->
        def v = (value as String).replace(',', ',\\\n')
        "$key=$v"
    }.join('\n\n')
    mkdir("$jlinkBasePath/META-INF")
    new File("$jlinkBasePath/META-INF/spring.factories").text = content

    // insert META-INF/spring.factories into the main jar
    ant.zip(update: "true", destfile: jar.archivePath, keepcompression: true) {
        fileset(dir: "$jlinkBasePath", includes: 'META-INF/**')
    }
}

jlink {
    imageZip = file("$buildDir/image-zip/registration-service-image.zip")
    options = ['--strip-debug', '--compress', '2', '--no-header-files', '--no-man-pages']
    forceMerge 'jaxb-api', 'byte-buddy', 'classgraph'
    mergedModule {

        uses 'ch.qos.logback.classic.spi.Configurator'

        excludeRequires 'com.fasterxml.jackson.module.paramnames'
        excludeProvides implementation: 'com.sun.xml.bind.v2.ContextFactory'
        excludeProvides servicePattern: 'javax.enterprise.inject.*'
        excludeProvides service: 'org.apache.logging.log4j.spi.Provider'
        excludeProvides servicePattern: 'reactor.blockhound.integration.*'
    }
    launcher {
        name = 'run'
        jvmArgs = [
                '--add-reads', 'registration.service.merged.module=com.sudhir.registration',
                '-cp', 'config/',
        ]
    }
}

tasks.jlink.doLast {
   
    copy {
        from "src/main/resources"
        into "$imageDir.asFile/bin/config"
    }

    copy {
        from "$buildDir/classes/java/main/com/sudhir/registration"
        into "$imageDir.asFile/bin/config/com/sudhir/registration/for-spring-classpath-scanner"
    }
}```


I just have one module called com.sudhir.registration. Inside this module, having just one RestController just to test the api.

**Comment by yadavsudhir405** [comment]
<img width="1680" alt="Screenshot 2020-07-14 at 11 31 19 PM" src="https://user-images.githubusercontent.com/11731706/87462098-0650ad80-c62d-11ea-9db2-dde878686272.png">

**Comment by yadavsudhir405** [comment]
This is how my module-info.java looks like:
```
open module com.sudhir.registration {
    requires spring.web;
    requires spring.beans;
    requires spring.boot;
    requires spring.boot.autoconfigure;

}
```

**Comment by siordache** [comment]
I don't know what is missing, but if you can put up your project on GitHub, I will try to find out.

**Comment by yadavsudhir405** [comment]
Hi @siordache  Thank you for your response.

Here is my link to sample [code](https://github.com/yadavsudhir405/registration-service-custom-image).

**Comment by siordache** [comment]
Your sample project works as expected on my machine: the "Hello World" message is displayed when I open http://localhost:8080/profile in a browser.

To troubleshoot the application, you can configure the DEBUG logging level in application.yml:
```
logging:
  level:
    root: DEBUG
    web: DEBUG
```

With this configuration, the application writes:
```
2020-07-15 08:03:16.322 DEBUG 13664 --- [nio-8080-exec-4] o.s.web.servlet.DispatcherServlet        : GET "/profile", parameters={}
2020-07-15 08:03:16.324 DEBUG 13664 --- [nio-8080-exec-4] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped to com.sudhir.registration.ProfileController#getProfile()
2020-07-15 08:03:16.331 DEBUG 13664 --- [nio-8080-exec-4] m.m.a.RequestResponseBodyMethodProcessor : Using 'text/html', given [text/html, application/xhtml+xml, image/webp, image/apng, application/xml;q=0.9, application/signed-exchange;v=b3;q=0.9, */*;q=0.8] and supported [text/plain, */*, text/plain, */*, application/json, application/*+json, application/json, application/*+json]
2020-07-15 08:03:16.332 DEBUG 13664 --- [nio-8080-exec-4] m.m.a.RequestResponseBodyMethodProcessor : Writing ["Hello World"]
2020-07-15 08:03:16.335 DEBUG 13664 --- [nio-8080-exec-4] o.s.web.servlet.DispatcherServlet        : Completed 200 OK
```


⚠️ In your sample project there is a typo in the name of the configuration file: `applicatiion.yml`.

**Comment by yadavsudhir405** [comment]
Thanks @siordache  for your help. I figured out the issues. Root cause was:  I was trying to run the image from another directory (say from my home directory) and the image file is stored in some other directory. Therefore, Spring was not able to scan RestController class because  classpath specified in build gradle file  is a relative path(config directory in my case). When I went to the bin directory of image-directory and run image file it worked.
